tutorials/intermediate/03-reinforcement-learning.ipynb

```diff
@@ -0,0 +1,801 @@
+      },
+      "source": [
+        "## Setting up the Environment\n",
+        "\n",
+        "Let's load our environment first."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "eTfYSODJTbux"
+      },
+      "source": [
+        "env = gym.make(\"CartPole-v0\")"
+      ],
+      "execution_count": 5,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "afRSrabwTq_a"
+      },
+      "source": [
+        "### On Colab\n",
+        "\n",
+        "If on Google Colab, we need to follow a list of steps to render the output. First we initialize our screen size."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "HbhZwa-JM7ub",
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "outputId": "223bdc5b-a057-40ee-8278-522ec1d9184b"
+      },
+      "source": [
+        "display = Display(visible=0, size=(1400, 900))\n",
+        "display.start()"
+      ],
+      "execution_count": 6,
+      "outputs": [
+        {
+          "output_type": "execute_result",
+          "data": {
+            "text/plain": [
+              "<pyvirtualdisplay.display.Display at 0x7f76f00bf810>"
+            ]
+          },
+          "metadata": {},
+          "execution_count": 6
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "l8qDfZi_R2xW"
+      },
+      "source": [
+        "Below we have a utility function to enable video recording of the gym environment. To enable video, we have to wrap our environment in this function."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "V2bZH987tHtT"
+      },
+      "source": [
+        "def wrap_env(env):\n",
+        "  env = Monitor(env, './video', force=True)\n",
+        "  return env\n",
+        "\n",
+        "env = wrap_env(env)"
+      ],
+      "execution_count": 7,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "_iOU0i6sUg9D"
+      },
+      "source": [
+        "### Set the seed"
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "colab": {
+          "base_uri": "https://localhost:8080/"
+        },
+        "id": "rQljT2b2k1Qy",
+        "outputId": "dfaa0eeb-e1c7-46c5-a266-a06b508229aa"
+      },
+      "source": [
+        "env.seed(seed_val)\n",
+        "torch.manual_seed(seed_val)"
+      ],
+      "execution_count": 8,
+      "outputs": [
+        {
+          "output_type": "execute_result",
+          "data": {
+            "text/plain": [
+              "<torch._C.Generator at 0x7f76fa684730>"
+            ]
+          },
+          "metadata": {},
+          "execution_count": 8
+        }
+      ]
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "ax8ToOAviGGS"
+      },
+      "source": [
+        "## Model\n",
```

Comment:
Let's explicitly tell that we solve this problem using Reinforce method, example ref: http://www.cs.toronto.edu/~tingwuwang/REINFORCE.pdf